### DISCUSSION TITLE
Set field in repeater

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#8268] [Help]
I've studied https://filamentphp.com/docs/2.x/forms/fields#using-get-to-access-parent-field-values very intensive, but I can not access field in repeater.

```
 Card::make()->label('Product')
                        ->schema([
                            Repeater::make('content')->label('Product')
                                ->relationship()
                                ->schema([
                                    Select::make('amount')
                             ])
                          ])
Select::make('amount_split')->numeric()
```

I can access field outside the repater from repeater. But I want access a field IN repeater from outside!

1. How can I set the default of amount_split to amount?
2. How can I set the default, if amount_split changes?

Thank you so much!

**Comment by danharrin** [comment]
The default is only used when the form is loaded. You should use `afterStateUpdated()` on the repeater select to `$set()` the amount_split

**Reply by photojo** [reply]
Thank you.
But afterStateUpdated() only works, when repeater select is updated. But it is just used to show the value. I will even add ->disabled()

Only the amount_split is "available". As a wrote before, I want to access the value from repeater select.

**Reply by danharrin** [reply]
Is `amount_split` editable by the user manually

**Reply by photojo** [reply]
In this example, only the amount_split shoud be editable. This already works. 
BUT: it would be fine, that this field "interacts" with the select in repeater. 
So the amount_split have maxValue of select in repeater. When amount_split increases, select amount in repeater should decrease.
Is that possible?

**Reply by danharrin** [reply]
I don't really understand the use case here 🤔

**Comment by photojo** [comment]
I'm so sorry for my description. I try it in short form again.

```
Card::make()->label('Product')
                        ->schema([
                            Repeater::make('content')->label('Product')
                                ->relationship()
                                ->schema([
                                    Select::make('amount')   **// value 15**
                             ])
                          ])
Select::make('amount_split')->numeric()  **// only this field is editable**
```

When _amount_split_ is set to 5, _amount_ should be set to 10! And _amount_split_ maxValue has to be 15 of course!

Is this possible?!

**Reply by danharrin** [reply]
Yeah I still don't understand, what is the real life use case? It feels super unconventional

**Comment by photojo** [comment]
Ok. Let me explane.

1. I've an order 1 with the product 1 with an certain amount, f.e. potatoes 20 tons. The truck is already on the road.
2. An other client on the trucks route needs just 2 tons of the 20. So, I've a new order.
3. Now I want so split the first order in a second one. I do this with ReplicateAction. This works fine. But I have to avoid an amount over 20 tons in input field in order two.

Everything in order 2 is like in order 1, but the amount of potatoes is different. But both orders together have 20 tons potatoes.

**Reply by photojo** [reply]
Much more easier: how can i access an value from repeater select from outside repeater?

**Reply by danharrin** [reply]
`$get('repeater_name')` gives you an array of repeater data, you can get the select from inside that

**Reply by mitsyara** [reply]
@danharrin how can I set the value inside a repeater field?
Let's say I have an "Order ID" field outside the Repeater.

In the Repeater, I had multiple input fields, one of which was "Name".

How can I pass in the "Order ID" field value to the "Repeater" > "Name" field?

I tried using the array like:
`$set( ['repeater_name']['repeater_key']['name'] , $value )`. 
and using paths like:
`$set('repeater_name/repeater_key/name', $value)`
But none of these seems to work.
